rfd/0173-terraform-machine-id.md

```diff
@@ -0,0 +1,303 @@
+audit log.
+
+### UX
+
+This improves the UX for the "existing advanced user" persona as they don't need to install and configure `tbot`
+anymore. This also unblocks support for runtimes where running `tbot` was not possible:
+e.g. [Terraform Cloud](https://github.com/gravitational/teleport/issues/26345).
+
+This greatly improves the UX of the "getting started" personas as the provider will hide all the complexity and shorten
+the time to value for IaC adoption. The
+whole [setup Terraform provider page](https://github.com/gravitational/teleport/blob/master/docs/pages/management/dynamic-resources/terraform-provider.mdx)
+becomes a 2-step guide: run `tsh login` and create the `main.tf`.
+
+Two documentation guides will be published:
+- Getting started with the Terraform Provider, explaining users how to run the provider locally
+- Running the Terraform Provider in CI/CD pipelines, explaining users how to run the provider in GitHub Actions, GitLab
+  CI, Circle CI, Spacelift, (and Terraform Cloud when we'll add support).
+
+The Getting started guide should provide instruction on how to connect to a local cluster with self-signed certificates
+(recover and trust the cluster cert with `SSL_CERT_FILE` instead of running with --insecure).
+
+### Observability
+
+The new Provider joining mechanisms will reuse existing Teleport APIs which are already emitting audit events and
+reporting Prometheus metrics.
+
+By its short-lived nature, the terraform process does not expose metrics.
+
+### Product usage
+
+To validate product adoption and join method usage we need some Telemetry, there are two possible approaches:
+
+#### Anonymous opt-in Telemetry client-side
+
+In this approach, telemetry would be opt-in and reuse the existing tbot start event and adding new fields:
+
+```protobuf
+syntax = "proto3";
+
+message TbotStartEvent {
+  enum RunMode {
+    RUN_MODE_UNSPECIFIED = 0;
+    RUN_MODE_ONE_SHOT = 1;
+    RUN_MODE_DAEMON = 2;
+    RUN_MODE_IN_PROCESS = 3;
+  }
+  RunMode run_mode = 1;
+  string version = 2;
+  string join_type = 3;
+  string helper = 4;
+  string helper_version = 5;
+  int32 destinations_other = 6;
+  int32 destinations_database = 7;
+  int32 destinations_kubernetes = 8;
+  int32 destinations_application = 9;
+  // process is the name of the process running tbot when run mode is "in process".
+  // The only value will be "terraform" but we can reuse this in kube with "operator".
+  string process = 10;
+}
+```
+
+Telemetry would be gated by the `TELEPORT_ANONYMOUS_TELEMETRY=1` environment variable.
+
+#### Auth-based Telemetry for Teleport customers
+
+Depending on their license, Teleport customers can have clusters dialing home and reporting usage.
+We can leverage this to report metrics from the auth. An option would be to add an annotation on the bot resource such as
+`teleport.dev/integration: terraform` or `teleport.dev/usage: terraform`. Potential values could be expanded as we add
```

Comment:
Annotation strategy would work fine for your bootstrapping, but if someone set up a Bot/Token themselves, we wouldn't necessarily be aware.